### ISSUE TITLE
Hardware requirements for FastAPI

### ISSUE THREAD

**Issue by ligantx** [tiangolo/fastapi#5516]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
not code related
```

### Description

I am running my web server in my Raspberry Pi 4 8GB RAM, and it's working fine. But now I want to run the same project in a  Raspberry Pi 4 2GB RAM (4GB and 8 GB) aren't available on the local market. 
Is there any possibility it wont be able to run sufficiently?

### Operating System

Linux

### Operating System Details

Raspberry pi OS 64bit

### FastAPI Version

0.85.0

### Python Version

3.10.6

**Comment by JarroVGIT** [comment]
That really depends on how heavy your application (and load) is. Is it containerized? Because if so, you can see how much it currently uses, and if that would fit a 2GB model. But typically; there is not really a minimum, it should work just fine

**Comment by iudeen** [comment]
It should be fine, as long as you don't handle too many requests or requests that are very large. 

I would recommend you to use a containerized approach to limit the resource usage. That way, you will have control over the host in case something goes wrong.

We have experimented deploying FastAPI on SBCs using K3s and when our resources demand increased, we added another node (another SBC) to it. Each node was 1 GB.

**Comment by tiangolo** [comment]
Thanks for the help here everyone!

Yep, there are no hard limits or minimums from the FastAPI side. It is itself quite small and not heavy, so most of the resource consumption would really be your own app, so it would depend mostly on your code.

**Comment by ligantx** [comment]
> That really depends on how heavy your application (and load) is. Is it containerized? Because if so, you can see how much it currently uses, and if that would fit a 2GB model. But typically; there is not really a minimum, it should work just fine

No, it is running without Docker or anything similar. Also, the load is too small because it will run in a local network and it will be like 4 clients max reading stuff from the Database..

> Thanks for the help here everyone!
Yep, there are no hard limits or minimums from the FastAPI side. It is itself quite small and not heavy, so most of the resource consumption would really be your own app, so it would depend mostly on your code.

Thanks for answering man and mostly for developing this wonderful software. It has one of the best documentation i've found and made me really wanna invest time learning this..

**Comment by tiangolo** [comment]
Cool! Then I would think you should be fine. 😁

And thanks for the kind words! 😊😁🚀

**Comment by tiangolo** [comment]
Great to hear! Thanks for reporting back. In that case, I think you could close the issue, right?
